applications/incompressible_navier_stokes/vortex/application.h

```diff
@@ -455,96 +456,90 @@ class Application : public ApplicationBase<dim, Number>
+              {
+                cell->face(f)->set_all_manifold_ids(1);
+              }
+              else if(mesh_type == MeshType::ComplexVolumeManifold)
+              {
+                face_ids.push_back(f);
+                unsigned int manifold_id = manifold_ids.size() + 1;
+                cell->set_all_manifold_ids(manifold_id);
+                manifold_ids.push_back(manifold_id);
+                break;
+              }
+              else
+              {
+                AssertThrow(false, dealii::ExcMessage("Should not arrive here."));
+              }
+            }
           }
         }
       }
 
-      // generate vector of manifolds and apply manifold to all cells that have been marked
-      static std::vector<std::shared_ptr<dealii::Manifold<dim>>> manifold_vec;
-      manifold_vec.resize(manifold_ids.size());
-
-      for(unsigned int i = 0; i < manifold_ids.size(); ++i)
+      // set manifolds
+      if(mesh_type == MeshType::ComplexSurfaceManifold)
+      {
+        static const dealii::SphericalManifold<dim> spherical_manifold;
+        this->grid->triangulation->set_manifold(1, spherical_manifold);
+      }
+      else if(mesh_type == MeshType::ComplexVolumeManifold)
       {
-        for(auto cell : this->grid->triangulation->active_cell_iterators())
+        // generate vector of manifolds and apply manifold to all cells that have been marked
+        static std::vector<std::shared_ptr<dealii::Manifold<dim>>> manifold_vec;
+        manifold_vec.resize(manifold_ids.size());
+
+        for(unsigned int i = 0; i < manifold_ids.size(); ++i)
         {
-          if(cell->manifold_id() == manifold_ids[i])
+          for(auto cell : tria_coarse.active_cell_iterators())
           {
-            manifold_vec[i] =
-              std::shared_ptr<dealii::Manifold<dim>>(static_cast<dealii::Manifold<dim> *>(
-                new OneSidedCylindricalManifold<dim>(cell, face_ids[i], center)));
-            this->grid->triangulation->set_manifold(manifold_ids[i], *(manifold_vec[i]));
+            if(cell->manifold_id() == manifold_ids[i])
+            {
+              dealii::Point<dim> center = dealii::Point<dim>();
+              manifold_vec[i] =
+                std::shared_ptr<dealii::Manifold<dim>>(static_cast<dealii::Manifold<dim> *>(
+                  new OneSidedCylindricalManifold<dim>(cell, face_ids[i], center)));
+              tria_coarse.set_manifold(manifold_ids[i], *(manifold_vec[i]));
+            }
           }
         }
       }
+      else
+      {
+        AssertThrow(false, dealii::ExcMessage("Should not arrive here."));
+      }
 
-      // refine globally due to boundary conditions for vortex problem
-      this->grid->triangulation->refine_global(1);
+      tria_coarse.refine_global(1);
+      // make sure that the triangulation does not have refinements at this point
+      dealii::GridGenerator::flatten_triangulation(tria_coarse, *this->grid->triangulation);
```

Comment:
@kronbichler, @peterrum: I encountered a problem for this test case if `mesh_type == MeshType::ComplexVolumeManifold` or `mesh_type == MeshType::ComplexSurfaceManifold`. The setup of the Chebyshev coarse grid solver fails (which is not related to the changes regarding manifolds IDs proposed in this PR, the problem also occurs with the previous code version). Due to the boundary conditions for this "vortex problem" (we have both Dirichlet and Neumann on each of the four sides of the rectangular domain), we have  to refine once before setting the boundary IDs. However, this refinement is considered an additional level in multigrid. I assume the boundary conditions are then somewhat undefined for the coarse grid problem (???, what does the deal.II code do in such a case?), so that the Chebyshev coarse grid solver can not converge ...

So I tried to solve this by removing this refinement level by calling `flatten_triangulation()`, considering the resulting (already refined) triangulation as the coarse grid problem. The solver now indeed converges.

However, with `flatten_triangulation()`, the manifolds have also disappeared, but the manifolds are main purpose of this code!? Applying the manifolds after `flatten_triangulation()` for the coarse grid (which is already a "refined" one in terms of number of cells) would be more complicated since manifolds would also have to be applied to cells not located at the spherical internal boundary to achieve the same behavior as before. 

@kronbichler, @peterrum: If you also do not know a solution, I would suggest to remove this code dealing with `mesh_type == MeshType::ComplexVolumeManifold` and `mesh_type == MeshType::ComplexSurfaceManifold` completely from this test case, implying that we would loose some functionality.

Reply:
> Not sure. But cannot you attach the manifolds again after the flattening?

The problem is not the flattening, but that the grid is now finer. Since we apply the manifolds manually for all the relevant cells, this is much more complicated if the grid is refined. Using the coarsest possible grid, the criterion whether the cell touches the spherical internal boundary is sufficient as a criterion to apply the manifolds. This does no longer work when using the refined grid.

Reply:
@kronbichler what do you think?